src/consensus/role_leader.rs

```diff
@@ -93,13 +93,16 @@ impl LeaderRole for Consensus {
             self.bft_round_state.staking
         );
 
-        let cut = match self
-            .bus
-            .request(QueryNewCut(self.bft_round_state.staking.clone()))
-            .await
+        let cut = match tokio::time::timeout(
+            std::time::Duration::from_millis(self.config.consensus.slot_duration / 2),
```

Comment:
I think it's too restrictive. We have more time than a half slot to do a cut before being timedout

Reply:
Would the full slot_duration work ?